diff.py

```diff
@@ -2188,7 +2245,16 @@ def diff_sameline(old: str, new: str) -> None:
         if ignore_last_field:
             newfields = newfields[:-1]
             oldfields = oldfields[:-1]
+        else:
+            # If the last field has a register suffix, i.e. "0x38(r7)"  we split that part out to make it a separate field
+            newfields = newfields[:-1] + newfields[-1].split("(")
+            oldfields = oldfields[:-1] + oldfields[-1].split("(")
```

Comment:
The comment I made before about MIPS is still relevant. MIPS use `%lo(...)` syntax for relocs; if you split on `(` you'll butcher that. This means that e.g.:
- `lui a0, %hi(sym)` vs `lui a0, 0x8000` will split up as `["lui a0", " %hi", "sym)"]` vs `["lui a0", " 0x8000"]`, zip() will ignore the third field of the first, compare " %hi" against " 0x8000" and get a mismatch, and "penalize any extra fields" with add even more to the score. Which is kinda okay as an end result, but the path there is dubious and platform dependent
- `lui a0, %hi(sym)` vs `lui a0, %hi(.rodata + 0x10)` will almost break, because the %hi is split up so that the `imm_matches_everything` logic never finds a `(.`. Except it won't, because `lo_hi_match` takes care of that in advance as an ugly MIPS-specific hack.
- `lui a0, %hi(sym)` vs `lui a1, %hi(.rodata + 0x10)` *will* break, because `lo_hi_match` doesn't treat that case

Basically, this kinda-sorta mostly works, but it's hacky and we're spreading logic between here and `lo_hi_match`, dealing differently with different architectures. There's an opportunity here to get rid of `lo_hi_match` altogether in favor of `imm_matches_everything`. We need to fix this to not catch `(` that are part of %lo/%hi for MIPS though. Ideally we'd have a structured representation of the lines that would make that trivial, but lacking that... how about splitting on a regex that looks for `(` but with a negative lookbehind for `%lo/%hi`?

Reply:
> Which is kinda okay as an end result, but the path there is dubious and platform dependent

Yes, I analyzed this exact case, and landed that the behavior / end result is the desired behavior.. but I agree the path there is a little dubious.   But won't it result in correct behavior on all platforms?

>lui a0, %hi(sym) vs lui a0, %hi(.rodata + 0x10) will almost break, because the %hi is split up so that the >imm_matches_everything logic never finds a (.

Yea, so I feel like we need correct wildcard symbol catching for each arch... just like for ppc we check for `...data`...  We'd have a similar check here for .rodata + offset, or .data + offset for mips.... I only put it to "(." because I thought I remember a comment you had that said "(." is fine for mips, and I just went with that... maybe I misunderstood.

> will break, because lo_hi_match doesn't treat that case

Same as above.. we just need the correct wildcard symbol catching for MIPS like I worked out for PPC. 

>Basically, this kinda-sorta mostly works, but it's hacky and we're spreading logic between here and lo_hi_match, dealing differently with different architectures. There's an opportunity here to get rid of lo_hi_match altogether in favor of imm_matches_everything.

Yes, makes sense, it makes sense to me get rid of lo_hi_match, and just focus on getting imm_matches_Everything working for all archs..

>We need to fix this to not catch ( that are part of %lo/%hi for MIPS though. Ideally we'd have a structured representation of the lines that would make that trivial, but lacking that... how about splitting on a regex that looks for ( but with a negative lookbehind for %lo/%hi

I am confused why we need this.
In general, I feel like it makes sense to score and check separately for whats in the parenthesis and what is before it. (penalize one or the other, or both that is) And that goes for Mips and for ppc.  We just need to get make a good function for each arch, that detects the wild cards correctly right?

Unless, I am missing something, can you show me a case where splitting at the "(" will not work? assuming we have a good mips function for checking for wildcard symbols.

Reply:
Related question

```
    if arch.name == "ppc":
        if "...data" in field:
            return True
```

Do you know if ppc needs to check for more data sections other than just .data... like rodata, sdata, etc... I could never find any examples of where it would.

Reply:
I am changing mips back to the regex, and now it should have the desired behavior on all the cases you proposed.

Reply:
looking at mips code... it seems like any `field` that contains a "." the decimal should be a match to any symbol.
Does that sound right? I don't see an opposing case, but haven't looked at a ton of mips asm.
Because that really simplifies thing if that's the case...

and even if it's not the case, I should be able to mod the code and/or regex to accurately detect.

Reply:
and for anything that is not mips and not ppc... it will just return False.. which results in the same behavior as it was before this PR... and will make it easy to add in future for other archs and we want to detect these wildcard(match to any symbol) symbols to improve scoring.

Reply:
If you want I can modify this part:
`oldfields = oldfields[:-1] + oldfields[-1].split("(")`
So that it also removes the closing parenthesis at the end. It's not needed for current archs and code, but might be a little more clear/clean

Reply:
%lo(...) is an entire immediate, it doesn't make sense to cut it off halfway, even if we can reason ourselves to concluding that it works okay in practice. Did you look into my suggestion of using a regex with negative lookbehind for %lo/%hi?

Reply:
No, never in compiler-generated code. And yeah, I'm hoping we can get rid of lo_hi_match. Not because it gets things wrong, but because it does the same thing as imm_matches_everything but in another manner.

Reply:
okay makes sense about getting rid of lo_hi_match.

But you are sure, you don't want to check for all those cases I listed above?

Otherwise I will proceed with updating the code to not split at "(" if there is a proceeding %lo%hi.

And also, we do want to split off and check the final register separately in cases of this right:
%lo(D_808DFEC0)(a1) vs %lo(D_808DFEC0)(a2)

Reply:
the negative look-behind for %lo/%hi seems to work... but also I think it would also work, with a negative lookahead (look-forward) for a decimal point, "."

Reply:
Yeah, we want to split %lo(D_808DFEC0)(a1) in two. So negative lookahead for `.` wouldn't work